libs/platform/src/lib/components/list/list.component.ts

```diff
@@ -0,0 +1,647 @@
+    noSeperator: boolean;
+
+    /** The type of the selection. Types include:
+    *'none'| 'multi' | 'single'|'delete'.
+    * Leave empty for default ().'
+    * Default value is set to ''
+    */
+    @Input()
+    selectionMode: SelectionType = 'none';
+
+    /**ListType 'inactive' | 'active' | 'navigation' | 'detail' */
+    @Input()
+    listType: ListType = 'active';
+
+    /**  An array that holds a list of all selected items**/
+    @Input()
+    selectedItems: BaseListItem[];
+
+    /** define size of items for screen reader */
+    @Input()
+    ariaSetsize: number;
+
+    /** Defines whether items are multiseletable for screen reader */
+    @Input()
+    ariaMultiselectable: boolean;
+
+    /**Title used on button when data loads on button click */
+    @Input()
+    loadTitle: string;
+
+    /**
+   * Todo: Name of the entity for which DataProvider will be loaded.
+   *  You can either pass list of
+   * items or use this entityClass and internally we should be able
+   * to do lookup to some registry
+   * and retrieve the best matching DataProvider that is set on application level
+   *
+   *
+   */
+    @Input()
+    entityClass: string;
+
+
+    /**
+      * Child items of the List.
+      */
+    @ContentChildren(BaseListItem, { descendants: true }) ListItems: QueryList<BaseListItem>;
+
+    @ContentChild(ListItemDef) listItemDef: ListItemDef;
+
+    /** Access child element, for checking link content*/
+    @ViewChild('link', { read: ElementRef })
+    anchor: ElementRef;
+
+    /**
+     * @hidden
+     * keyManger to handle keybord events
+     */
+    keyManager: FocusKeyManager<BaseListItem>;
+
+    /**
+     * Whether Navigation mode is included to list component
+     * for all the items
+    */
+    navigated: boolean;
+
+    /**
+     * Whether Navigation mode is included to list component
+     * only a subset of the list items are navigable
+     * you should indicate those by displaying a navigation arrow
+    */
+    navigationIndicator: boolean;
+
+    /**
+     * Whether By line is present in list item*/
+    hasByLine: boolean;
+
+    /**@hidden
+     * To display loading symbol */
+    loading = false;
+
+    /** The model backing of the component. */
+    selectionModel: SelectionModel<BaseListItem>;
+
+    /**@hidden
+     * Whether list component has multiselection */
+    multiSelect = false;
+
+    /** @hidden
+    * To store */
+    tempItems = [];
+
+    startIndex = 0;
+
+    lastIndex = this.itemSize;
+
+    items = [];
+
+    dsItems = [];
+
+    selectedvalue: string;
+
+    /** @hidden */
+    _contentDensity = this._listConfig.contentDensity;
+
+    /**
+     * @hidden
+     * Used to define if contentDensity value is 'compact' or not.
+     */
+    isCompact = this._contentDensity === 'compact';
+
+    /** role */
+    @HostBinding('attr.role')
+    role = 'list';
+
+    _destroyed = new Subject<void>();
+    _dsSubscription: Subscription | null;
+    _itemsSubscription: Subscription | null;
```

Comment:
You can have one subscription `private _subscriptions = new Subscription();` and call `add()` method for all of subscriptions. Check dialog component in core for example

Reply:
_dsSubscription is used for handling datasource changes and _itemsSubscription for list item changes for such as multi-select and single select. 
Using add breaks the functionality.
Can not implement these changes considering the above issues.